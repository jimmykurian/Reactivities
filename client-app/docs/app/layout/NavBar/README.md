[**Client App Documentation v0.0.0**](../../../README.md) â€¢ **Docs**

***

[Client App Documentation v0.0.0](../../../README.md) / app/layout/NavBar

# app/layout/NavBar

## Index

### Interfaces

- [Props](interfaces/Props.md)

### Functions

- [default](functions/default.md)
