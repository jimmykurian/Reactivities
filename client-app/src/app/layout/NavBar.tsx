/**
 * @file NavBar.tsx
 * @author Jimmy Kurian
 * @fileoverview This file contains the NavBar component which serves as the navigation bar for the Reactivities application.
 */

import { Button, Container, Menu } from 'semantic-ui-react';
import { useStore } from '../stores/store';

/**
 * @component NavBar
 * @description The NavBar component serves as the navigation bar for the Reactivities application.
 * It includes a logo, a menu item for activities, and a button to create a new activity.
 *
 * @returns {JSX.Element} The rendered NavBar component.
 *
 * @example
 * ```tsx
 * import React from 'react';
 * import NavBar from './NavBar';
 *
 * function App() {
 *   return (
 *     <div>
 *       <NavBar />
 *       // other components
 *     </div>
 *   );
 * }
 *
 * export default App;
 * ```
 */
export default function NavBar(): JSX.Element {
  const { activityStore } = useStore();

  return (
    <Menu inverted fixed="top">
      <Container>
        <Menu.Item header>
          <img
            src="/assets/logo.png"
            alt="logo"
            style={{ marginRight: '10px' }}
          />
          Reactivities
        </Menu.Item>
        <Menu.Item name="Activities" role="menuitem" />
        <Menu.Item>
          <Button
            onClick={() => activityStore.openForm()}
            positive
            content="Create Activity"
          />
        </Menu.Item>
      </Container>
    </Menu>
  );
}
