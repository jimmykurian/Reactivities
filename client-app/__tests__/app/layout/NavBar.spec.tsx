import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import NavBar from '../../../src/app/layout/NavBar';

describe('NavBar', () => {
  test('renders the logo image', () => {
    render(<NavBar />);
    const logoImage = screen.getByAltText('logo');
    expect(logoImage).toBeInTheDocument();
    expect(logoImage).toHaveAttribute('src', '/assets/logo.png');
  });

  test('renders the header title', () => {
    render(<NavBar />);
    const headerTitle = screen.getByText(/Reactivities/i);
    expect(headerTitle).toBeInTheDocument();
  });

  test('renders the Activities menu item', () => {
    render(<NavBar />);
    const activitiesMenuItem = screen.getByRole('menuitem', {
      name: /Activities/i,
    });
    expect(activitiesMenuItem).toBeInTheDocument();
  });

  test('renders the Create Activity button', () => {
    render(<NavBar />);
    const createActivityButton = screen.getByRole('button', {
      name: /Create Activity/i,
    });
    expect(createActivityButton).toBeInTheDocument();
    expect(createActivityButton).toHaveClass('positive');
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
