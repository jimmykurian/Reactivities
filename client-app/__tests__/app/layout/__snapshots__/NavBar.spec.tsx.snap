// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui inverted top fixed menu"
  >
    <div
      class="ui container"
    >
      <div
        class="header item"
      >
        <img
          alt="logo"
          src="/assets/logo.png"
          style="margin-right: 10px;"
        />
        Reactivities
      </div>
      <div
        class="item"
        role="menuitem"
      >
        Activities
      </div>
      <div
        class="item"
      >
        <button
          class="ui positive button"
        >
          Create Activity
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
